src/model.cpp

```diff
@@ -1027,9 +1030,14 @@ void Model::fdydp(const realtype t, const AmiVector *x) {
 
     std::fill(dydp.begin(),dydp.end(),0.0);
     fw(t,x->data());
-    fdwdp(t,x->data());
+    auto dense = fdwdp(t,x->data());
+    // if dwdp is not dense, fdydp will expect the full sparse array
+    realtype *dwdp_tmp = dwdp.data();
     for(int ip = 0; ip < nplist(); ip++){
         // get dydp slice (ny) for current time and parameter
+        if (dense && nw > 0) // if dwdp is dense, fdydp will expect a slice
```

Comment:
Move if outside loop. Better 2 separate loops than conditional inside loop.

Reply:
cant move slicing out of loop :(